doc/glossary.rst

```diff
@@ -85,12 +128,30 @@ Glossary of Terms
       integrated circuit that are not in use.
 
    SoC
-      `System on a chip`_
+      A `System on a chip`_, that is, an integrated circuit that contains at
+      least one :term:`CPU cluster` (in turn with at least one :term:`CPU core`),
+      as well as peripherals and memory.
+
+   SoC family
+      One or more :term:`SoCs <SoC>` or :term:`SoC series` that share enough
```

Comment:
Should there be a term for extended board?

Reply:
Not sure what you mean by that?

Reply:
This feature was added some time ago https://github.com/zephyrproject-rtos/zephyr/pull/57390/files might make sense to have the term added to the glossary

Reply:
ooh right. The glossary needs tons of updates, among them this one :)  I will improve the glossary soon with another PR